Filling in Use Cases Document

eric-schuh opened the pull request:
To this point we have been focusing on the Focal Use Cases, and the sequence diagrams specifically. This is a first pass at the rest of this use cases document. It is largely based off of the VC Use Cases document with updates to reflect the split in roles for the Issuer and Verifier, as discussed in the VC-API group and updated use case descriptions to highlight this bifurcation in roles beyond what was discussed in the VCWG.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/eric-schuh/vc-api-use-cases/pull/24.html" title="Last updated on Jan 17, 2023, 8:45 PM UTC (68ee885)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api-use-cases/24/9c58bc5...eric-schuh:68ee885.html" title="Last updated on Jan 17, 2023, 8:45 PM UTC (68ee885)">Diff</a>

eric-schuh committed:
Update index.html

eric-schuh committed:
Added introduction

eric-schuh committed:
added VC use cases section as a starting point

eric-schuh committed:
updated front matter of use cases section

included the services and coordinators as a split of the Verifier and Issuer roles from the standard VCWG

eric-schuh committed:
updated use cases through health care section

eric-schuh committed:
updated the rest of the use cases

eric-schuh committed:
adding images for user roles and problem domain

eric-schuh committed:
fixing image links

jandrieu commented:
We should figure out how to clean up the ReSpec errors. The current respec preview rendering attempts (and fails) to link terms and other niceties, but currently generates errors, displayed with a squiggly red underline.

eric-schuh commented:
In the User Roles diagram, revert back to the VC Use Cases roles but have the split in verifier and issuer and "components".  This had a lot of support in Dec-12 call. Lets split the roles and components to make the mental model cleaner. A component is used to actualize a role, the VC-API needs to talk about components in addition to the roles

eric-schuh commented:
From Dec-12 Call: 
Ted - Does the "subject" belong as a role? Should the "holder" be split into a couple of different roles (issued to, or presenting to)? i.e. "Presenter" and "Issuee"

Manu - Three roles seems good, don't want to add complexity to the roles.

eric-schuh commented:
Remove capitalization from terms

eric-schuh commented:
Consider an "existing use" section where we can put real world deployments (see W3C CSV group for example of this type of section)

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
-        standard-governed way for Issuers, Holders and Verifiers of these credentials to transport them between said entities.
-        TODO: Finish introduction
+        Fundamentally, the Verifiable Credential API (VC-API) is a standard developed to give technologists a way to enable holders of
+        VCs to use them how and where they choose. Verifiable Credentials make claims which are a qualification, achievement, quality, or piece 
```
```suggestion
        VCs to use them how and where they choose. Verifiable Credentials make claims which may be a qualification, achievement, quality, or piece 
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        Fundamentally, the Verifiable Credential API (VC-API) is a standard developed to give technologists a way to enable holders of
+        VCs to use them how and where they choose. Verifiable Credentials make claims which are a qualification, achievement, quality, or piece 
+        of information about an entity's background such as a name, government ID, payment provider, home address, or university degree. 
+        The use cases outlined here are provided in order to make progress toward possible future standardization and interoperability of the transport
```
```suggestion
        The use cases outlined here are provided to help make progress toward possible future standardization and interoperability of the transport
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+  </section>
+  <section id='sotd'>
+      <p>
+        This document represents a concise but limited collection of use cases readers
+        should review alongside the <a href="https://w3c-ccg.github.io/vc-api/">Verifiable
```
```suggestion
        This document represents a concise but limited collection of use cases meant
        to be reviewed alongside the <a href="https://w3c-ccg.github.io/vc-api/">Verifiable
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        Credentials API</a>.
+      </p>
+      <p>
+        The work on this document is ongoing and subject to change as the conversation around the
```
```suggestion
        Work on this document is ongoing and subject to change as the conversation around the
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+      <p>
+        The work on this document is ongoing and subject to change as the conversation around the
+        VC API evolves in the main group. This should currently be treated as a draft document and
+        as such will have errors. As the VC API gets closer to a completed specification this 
+        document should reflect the current understanding and status of that work. 
```
```suggestion
        as such will have errors. As the VC API gets closer to a completed specification, this 
        document should reflect the current understanding and status of that work. 
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
-      <h3>Existing Work</h3>
+      <h3>Importance of this work</h3>
+        <p>
+          Entities (people, organizations, devices) need to make many kinds of
```
```suggestion
          <a>Entities</a> (people, organizations, devices) need to make many kinds of
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        production software on the internet works, for the purposes of the VC-API there arises
+        the need to delve into some detail as to how the Issuer and Verifier roles are structured.
+        This is due to the fact that there are many use cases where one company is making use of 
+        services to handle part of the technology stack for them. This has lead to both Verifier
```
```suggestion
        services to handle part of the technology stack. This has lead to both Verifier
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        the need to delve into some detail as to how the Issuer and Verifier roles are structured.
+        This is due to the fact that there are many use cases where one company is making use of 
+        services to handle part of the technology stack for them. This has lead to both Verifier
+        and Issuer splitting into two entities each: <a>Issuer Service</a>, <a>Issuer Coordinator</a>
```
```suggestion
        and Issuer splitting into two entities each: <a>Issuer Service</a>, <a>Issuer Coordinator</a>,
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        <a>Verifier Service</a>, and <a>Verifier Coordinator</a>. 
+      </p>
+      <p>
+        This leads to the full set of roles being considered in the VC-API as: 
```
```suggestion
        Thus, the full set of roles considered in the VC-API are
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+      <p>
+        There are four roles supported by <a>Verifiable Credentials</a> as defined by the 
+        VCWG: <a>Issuer</a>,
+        <a>Verifier</a>, <a>Subject</a>, and <a>Holder</a>.
```
```suggestion
        <a>Verifier</a>, and <a>Holder</a>.
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+      <p>
+        This leads to the full set of roles being considered in the VC-API as: 
+        <a>Issuer Service</a>, <a>Issuer Coordinator</a>, <a>Verifier Service</a>, <a>Verifier Coordinator</a>,
+        <a>Subject</a>, and <a>Holder</a>.
```
```suggestion
        and <a>Holder</a>.
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        Issuer Service
+      </dt>
+      <dd>
+        The <a>entity</a> that creates a <a>claim</a> and associates it with a
+        particular <a>subject</a>.
```
I don't think this is correct. As I understand it, the Issuer Service (software) delivers a VC, which contains the claims made by the Issuer Coordinator (human, possibly plus software) about one or more subjects.

eric-schuh replied:
Updated language to reflect this, new language is in the following commit:

772a31769b8263ff00adbb591110c2e31c7ce198

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+        Verifier Service
+      </dt>
+      <dd>
+        The <a>entity</a> performing the verifaction of a <a>claim</a> about a given <a>subject</a>.
```
```suggestion
        The <a>entity</a> performing the verification of a <a>Verifiable Credential</a> containing <a>claims</a> about a given <a>subject</a>.
```

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+      </dt>
+      <dd>
+        The <a>entity</a> that deals with business rules and policies around a
+        user's request for a particular <a>claim</a> to be verified. 
```
```suggestion
        user's request for a particular <a>Verifiable Credential</a> to be verified. 
```

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

TallTed commented on index.html:
```diff
@@ -152,45 +152,747 @@
+      <p>
+        There are four roles supported by <a>Verifiable Credentials</a> as defined by the 
+        VCWG: <a>Issuer</a>,
+        <a>Verifier</a>, and <a>Holder</a>.
```
Either the number changes as I have, or there's another role to name in the list...
```suggestion
        There are three roles supported by <a>Verifiable Credentials</a> as defined by the 
        VCWG: <a>Issuer</a>,
        <a>Verifier</a>, and <a>Holder</a>.
```

eric-schuh committed:
updating language around Issuer Service definition

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Split roles into components

Updated language and added figure to split the Verifier and Issuer Roles into components rather then adding new roles to the VCWG user model

eric-schuh commented:
![role-components](https://user-images.githubusercontent.com/22247137/211654359-baa31d51-ad1e-4875-ba6b-77bed6857da8.png)
![user-roles](https://user-images.githubusercontent.com/22247137/211655196-d1859f93-caa5-4906-87c0-09a446d096ac.png)

New images may need to be changed later, can't figure out how to change the images in the PR files listing or add new ones.

eric-schuh commented:
@jandrieu ready for review, I may have to add new images after this PR is accepted since I can't find a way to change the images in the PR itself. Once done I will take care of images.

Also I looked into fixing the respec but it is not clear what is wrong with the terms.html call. Everything looks identical to the vc-use-cases in the index.html when relating to terms so will need to dig into that more later.

TallTed commented on index.html:
```diff
@@ -240,43 +240,54 @@ <h3>Use case model</h3>
-        VCWG: <a>Issuer</a>,
-        <a>Verifier</a>, and <a>Holder</a>.
+        There are three roles supported by <a>Verifiable Credentials</a> as defined by the 
+        VCWG: <a>Issuer</a>, <a>Verifier</a>, and <a>Holder</a>. A fourth role is defined in
+        the VC Use Cases, this is the role of subject, however, for the purposes of the VC-API
+        the subject does not have an explicit role outside of acting as both Holder and Subject.
+        Do to this, the Subject role does not appear here in the VC-API. 
```
```suggestion
        VCWG: <a>Issuer</a>, <a>Verifier</a>, and <a>Holder</a>. In the VC Use Cases, a fourth
        role — that of subject — is defined. However, for the purposes of the VC-API, the subject
        is not relevant except when the same entity is both the Holder and the subject. For this
        reason, the Subject role does not appear in the VC-API.
```

TallTed commented on index.html:
```diff
@@ -240,43 +240,54 @@ <h3>Use case model</h3>
         production software on the internet works, for the purposes of the VC-API there arises
         the need to delve into some detail as to how the Issuer and Verifier roles are structured.
         This is due to the fact that there are many use cases where one company is making use of 
         services to handle part of the technology stack. This has lead to both Verifier
```
```suggestion
        services to handle part of the technology stack. This has led to both Verifier
```

TallTed commented on index.html:
```diff
@@ -240,43 +240,54 @@ <h3>Use case model</h3>
         </figcaption>
       </figure>
+      <p>
+        The combination of the respective Service and Coordinator components combine to fulfill
```
```suggestion
        The respective Service and Coordinator components combine to fulfill
```

TallTed commented on index.html:
```diff
@@ -240,43 +240,54 @@ <h3>Use case model</h3>
+      <p>
+        The combination of the respective Service and Coordinator components combine to fulfill
+        the role of the Issuer and Verifier as described in the VCWG. 
+        It is fully expected by the VC-API that software will exist that may fulfil only part of
+        each of the roles by implimenting any combnation of these components: 
```
```suggestion
        It is fully expected by the VC-API that software will exist that may fulfill only part of
        any of the roles, by implementing any combination of these component roles: 
```

TallTed commented on index.html:
```diff
@@ -240,43 +240,54 @@ <h3>Use case model</h3>
-        The <a>entity</a> that creates a <a>claim</a> and associates it with a
-        particular <a>subject</a>.
+        The <a>entity</a> that creates generates and delivers a <a>Verifiable Credential</a> to the Issuer Coordinator
+        which contains a some set of <a>claims</a> made by the Issuer Coordinator about one or more subjects. 
```
```suggestion
        The <a>entity</a> that creates, generates, and delivers a <a>Verifiable Credential</a> — which contains
        a some set of <a>claims</a> made by the Issuer Coordinator about one or more subjects — to the Issuer Coordinator. 
```

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh committed:
Update index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

eric-schuh commented:
> 

On the respec errors, I believe most of the terminology errors are due to the PR system. The same terminology section in the main repo properly grabs the terms and the majority of red underlines are not seen. I think I would like to just get this merged and see what set of respec issues still exist then fix those in a new PR since currently it is hard to say which are from the PR system and which are real (at least to me)

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        Issuer Service
+      </dt>
+      <dd>
+        The <a>entity</a> that creates, generates, and delivers a <a>Verifiable Credential</a> — which contains
```
```suggestion
        The <a>component</a> that creates, generates, and delivers a <a>Verifiable Credential</a> — which contains
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        Issuer Coordinator
+      </dt>
+      <dd>
+        The <a>entity</a> that deals with business rules and policies around a
```
```suggestion
        The <a>component</a> that deals with business rules and policies around a
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        Verifier Service
+      </dt>
+      <dd>
+        The <a>entity</a> performing the verification of a <a>Verifiable Credential</a> containing <a>claims</a> about a given <a>subject</a>.
```
```suggestion
        The <a>component</a> performing the verification of a <a>Verifiable Credential</a> containing <a>claims</a> about a given <a>subject</a>.
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        Verifier Coordinator
+      </dt>
+      <dd>
+        The <a>entity</a> that deals with business rules and policies around a
```
```suggestion
        The <a>component</a> that deals with business rules and policies around a
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        Subject
+      </dt>
+      <dd>
+        The <a>entity</a> about whom a <a>claim</a> is issued.
```
```suggestion
        The <a>entity</a> about whom a <a>claim</a> is made.
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          <a>credential repository</a> service to all students and alumni, so he chooses
+          to use it. In his third year, Rocky decides to transfer to Moosylvania Tech.
+          They do not offer a service, but he does not want to continue to use the
+          service of his old (and now rival school) so he moves his <a>claims</a> to the
```
```suggestion
          service of his old (and now rival school) so he moves his <a>verfifiable credentials</a> to the
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          Company wants to be certain that his shipping address is accurate (inaccurate
+          addresses are very expensive in terms of customer service). They offer a
+          discount for customers who make verifiable addresses available as part of the
+          checkout process. Francis offers his certificate to Giant Shoe Company's 
```
```suggestion
          checkout process. Francis offers a verifiable credential--containing a claim by a known authority that a specific address has been verified--to Giant Shoe Company's 
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        </dt>
+        <dd>
+          June goes to her local beer and wine store to buy a bottle of wine. She submits
+          her identity credential that lets the liquor store owner know that she is over
```
```suggestion
          a proof-of-age credential that lets the liquor store owner know that she is over
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          Mondays. So before actually purchasing a gift from the web shop of his choice,
+          he requests the shop to provide a credential issued by the chamber of commerce,
+          that contains proof of legitimacy. After having verified that the shop is
+          legit, by making use of the Wallet Verifier Service built into his
```
```suggestion
          legit, by making use of the Verifier Service built into his
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          to the bank website, logs in, and navigates to the KYC page. She then
+          selects the government-supplied <a>verifiable credentials</a> that confirm she
+          receives postal mail at a certain address and that she has a national ID card,
+          submitting these. MidBank has stood up their own verification service, which 
```
```suggestion
          receives postal mail at a certain address and that she has a national ID card.
          She submits these to Midbank's verification coordinator. MidBank has their own verification service, which 
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        <dd>
+          Alice wants to open a new bank account. BigOnlineBank offers the ability to
+          do this from home if she can provide electronic credentials.  She offers
+          government-issued certificates that verify her identity (address, national
```
```suggestion
          government-issued certificates that verify her attributes (address, national
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          The state's board makes use of the international Medical Certificate Issuer
+          to issue credentials that adhere to the international standards for 
+          medical practitioners. Upon completing his state's board certificate test, 
+          Barney is issued a digital certificate confirming that he is
```
```suggestion
          Barney is issued a verifiable credential confirming that he is
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          medical practitioners. Upon completing his state's board certificate test, 
+          Barney is issued a digital certificate confirming that he is
+          certified to practice medicine in that state.  Barney can now use this
+          certificate when writing prescriptions and referrals, thereby improving
```
```suggestion
          credential when writing prescriptions and referrals, thereby improving
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          through her health care network and goes in for treatment. She is a new
+          patient, so the clinic needs to know who she is and how she will be paying.
+          When checking in, she presents her <a>verifiable credential</a> that
+          demonstrates her identity and her proof of insurance, which is checked by
```
```suggestion
          demonstrates her identity attributes and her proof of insurance, which is checked by
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+        <dd>
+          Jane was issued a certificate by BigTraining Co., indicating that she was a
+          trained Project Manager.  It was later discovered that BigTraining Co. was not
+          actually training anyone, and their organization's certificate was revoked via
```
```suggestion
          actually training anyone, and their organization's authority to issue Project Manager certifications was revoked via
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          trained Project Manager.  It was later discovered that BigTraining Co. was not
+          actually training anyone, and their organization's certificate was revoked via
+          the US Department of Education's Accreditation Database. Jane's credential is
+          therefore invalid, and when Jane goes to submit the credentail received from
```
```suggestion
          therefore invalid, and when Jane goes to submit the credential received from
```

jandrieu commented on index.html:
```diff
@@ -152,45 +152,758 @@
+          social network to make herself readily available for new opportunities in the
+          workforce. She lists her employment history and credentials including degrees,
+          certificates, and digital badges. The website requests verification of her
+          credential <a>claims</a> in order for her credentials to be visible when she
```
```suggestion
          <a>claims</a> in order for her credentials to be visible when she
```

jandrieu reviewed (changes requested):
Overall, this is a great update. I've suggested some clean up. We also might think through how to be more rigorous in our use of credentials v certifications and certificates. There's a lot of confusion about how a real-world certification is represented in a VC and it would be good to tease out language to delineate in the use cases when it is the VC that we are talking about verses the underlying certification.

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh committed:
Update index.html

Co-authored-by: Joe Andrieu <joe@andrieu.net>

eric-schuh commented:
> Overall, this is a great update. I've suggested some clean up. We also might think through how to be more rigorous in our use of credentials v certifications and certificates. There's a lot of confusion about how a real-world certification is represented in a VC and it would be good to tease out language to delineate in the use cases when it is the VC that we are talking about verses the underlying certification.

Added this note as an issue for next chunk of work. Merging this PR after accepting the general text changes made along with this review
